gnmidiff/setrequest.go

```diff
@@ -0,0 +1,149 @@
+// Copyright 2023 Google Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// gnmidiff contains gNMI utilities for diffing SetRequests and GetResponses.
+package gnmidiff
+
+import (
+	"fmt"
+
+	"github.com/derekparker/trie"
+	"github.com/openconfig/ygot/ygot"
+
+	gpb "github.com/openconfig/gnmi/proto/gnmi"
+)
+
+type setRequestIntent struct {
+	// Deletes are deletions to any node.
+	Deletes map[string]struct{}
```

Comment:
do these fields need to be exported?  if the struct isnt

Reply:
Is there a guideline on this? I just wanted to make it capitalized in case one day we do want to export it.

Reply:
i'm not aware of any, generally i prefer not exported things until they are needed. if the plan is to export this in the future this is fine as is

Reply:
I will export it in a later PR.